core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -121,10 +125,77 @@ protected void doStop() throws Exception {
+        config.getString(RestConfig.REST_PROTOCOL_CONFIG).equals(RestConfig.REST_PROTOCOL_HTTP_PLUS_HTTPS)) {
+      SslContextFactory sslContextFactory = new SslContextFactory();
+      // IMPORTANT: the key's CN, stored in the keystore, must match the FQDN. This is a Jetty requirement.
+      if (!config.getString(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG).isEmpty()) {
+        sslContextFactory.setKeyStorePath(config.getString(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG));
+        sslContextFactory.setKeyStorePassword(config.getString(RestConfig.SSL_KEYSTORE_PASSWORD_CONFIG));
+        sslContextFactory.setKeyManagerPassword(config.getString(RestConfig.SSL_KEY_PASSWORD_CONFIG));
+        sslContextFactory.setKeyStoreType(config.getString(RestConfig.SSL_KEYSTORE_TYPE_CONFIG));
+
+        if (!config.getString(RestConfig.SSL_KEYMANAGER_ALGORITHM_CONFIG).isEmpty()) {
+          sslContextFactory.setSslKeyManagerFactoryAlgorithm(
+                  config.getString(RestConfig.SSL_KEYMANAGER_ALGORITHM_CONFIG));
+        }
+      }
+
+      sslContextFactory.setNeedClientAuth(config.getBoolean(RestConfig.SSL_CLIENT_AUTH_CONFIG));
+
+      if (!config.getString(RestConfig.SSL_ENABLED_PROTOCOLS_CONFIG).isEmpty()) {
+        sslContextFactory.setIncludeProtocols(config.getString(RestConfig.SSL_ENABLED_PROTOCOLS_CONFIG).split(","));
+      }
+
+      if (!config.getString(RestConfig.SSL_CIPHER_SUITES_CONFIG).isEmpty()) {
+        sslContextFactory.setIncludeCipherSuites(config.getString(RestConfig.SSL_CIPHER_SUITES_CONFIG).split(","));
+      }
+
+      if (!config.getString(RestConfig.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG).isEmpty()) {
+        sslContextFactory.setEndpointIdentificationAlgorithm(
+                config.getString(RestConfig.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG));
+      }
+
+      if (!config.getString(RestConfig.SSL_TRUSTSTORE_LOCATION_CONFIG).isEmpty()) {
+        sslContextFactory.setTrustStorePath(config.getString(RestConfig.SSL_TRUSTSTORE_LOCATION_CONFIG));
+        sslContextFactory.setTrustStorePassword(config.getString(RestConfig.SSL_TRUSTSTORE_PASSWORD_CONFIG));
+        sslContextFactory.setTrustStoreType(config.getString(RestConfig.SSL_TRUSTSTORE_TYPE_CONFIG));
+
+        if (!config.getString(RestConfig.SSL_TRUSTMANAGER_ALGORITHM_CONFIG).isEmpty()) {
+          sslContextFactory.setTrustManagerFactoryAlgorithm(
+                  config.getString(RestConfig.SSL_TRUSTMANAGER_ALGORITHM_CONFIG));
+        }
+      }
+
+      sslContextFactory.setProtocol(config.getString(RestConfig.SSL_PROTOCOL_CONFIG));
+      if (!config.getString(RestConfig.SSL_PROVIDER_CONFIG).isEmpty()) {
+        sslContextFactory.setProtocol(config.getString(RestConfig.SSL_PROVIDER_CONFIG));
+      }
+
+      NetworkTrafficServerConnector sslConnector = new NetworkTrafficServerConnector(server, sslContextFactory);
+      sslConnector.addNetworkTrafficListener(new MetricsListener(metrics, "jetty-https", metricTags));
```

Comment:
They are actually identical because they use the same underlying metrics tags LinkedList. I have a test ensuring that they're identical.